DownFall/Tutorial/TutorialModel.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  TutorialModel.swift
+//  DownFall
+//
+//  Created by William Katz on 11/26/19.
+//  Copyright © 2019 William Katz LLC. All rights reserved.
+//
+
+import Foundation
+
+extension IndexPath {
+    mutating func incrStepIndex() {
+        self = IndexPath(item: item+1, section: section)
+    }
+
+    mutating func incrSectionIndex() {
+        self = IndexPath(item: item, section: section+1)
+    }
+}
+
+class TutorialData {
+    let sections: [TutorialSection]
+    var currentIndexPath: IndexPath = IndexPath(item: 0, section: 0)
+    
+    init(sections: [TutorialSection]) {
+        self.sections = sections
+    }
+    
+    var currentStep: TutorialStep {
+        return step(sectionIndex: currentIndexPath.section, stepIndex: currentIndexPath.item)
+    }
+    
+    var currentSection: Int {
+        return currentIndexPath.section
+    }
+    
+    func incrSectionIndex() {
+        //make sure we dont go out of bounds
+        if currentSection < sections.count - 1 {
+            currentIndexPath.incrSectionIndex()
+            currentIndexPath = IndexPath(item: 0, section: currentIndexPath.section)
+        }
+    }
+    
+    func incrStepIndex() {
+        currentIndexPath.incrStepIndex()
+        if currentIndexPath.item >= countStepsIn(sectionIndex: currentIndexPath.section) {
+            currentIndexPath = IndexPath(item: currentIndexPath.item - 1, section: currentIndexPath.section)
```

Comment:
We need to leave a comment here or else I will never remember.  better yet we need to  add test before pushing this code.  

We should check that TutorialData and correctly move from step to step and then section to section.  We also never want to go out of bounds, so check for all those cases.  The main case is when we reach the last step in the last section and cannot incr step or incr section

Reply:
Removed unused code and added test against TutorialData